### PR TITLE
Some more fixes

### PR DESCRIPTION
**Pull request by oscargus**
Small change of behaviour in LabelPatternUtil.

However, I think that the previous code didn't even work... How can e.g.`"univ".equalsIgnoreCase(k.substring(0, 5))` ever evaluate to true? The substring is five characters, right? This is something for static code checkers to find...

### CODE REVIEW

**Comment by oscargus** [comment]
The background of the regex-solution is the issues pointed out by Codacy, requiring a locale to do "proper" toLowerCase/toUpperCase. This doesn't solve every issue related to that, but at least a few... Shouldn't be that frequently used so performance is an issue.

**Comment by simonharrer** [comment]
Why not just pass Locale.EN to the to*Case methods?

**Comment by oscargus** [comment]
Yes, that is of course a simple solution. :-) However, there's no need (in the general case) to support other languages? Considering that we still have Unicodesupport. What happens to ß?

**Comment by simonharrer** [comment]
yes, but when comparing it to a constant later, it is not that much of an issue, I think.

**Comment by simonharrer** [comment]
Could you please update this and just use toLowerCase with Locale.EN? Then, this can be merged in.

**Comment by oscargus** [comment]
Changed back (but kept the uni/dep/etc detection, which IMHO is better now).

File: src/main/java/net/sf/jabref/importer/fileformat/BibtexParser.java [hunk 1]
```diff
@@ -616,8 +616,7 @@ private String parseTextToken() throws IOException {
                 return token.toString();
             }
 
-            if (Character.isLetterOrDigit((char) character) || (character == ':') || (character == '-') || (character == '_')
-                    || (character == '*') || (character == '+') || (character == '.') || (character == '/') || (character == '\'')) {
+            if (Character.isLetterOrDigit((char) character) || (":-_*+./'".indexOf(character) >= 0)) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/importer/fileformat/BibtexParser.java]: Maybe replace with Arrays.asList(':', '-', .... ).contains(character). But I'm not sure which one is more readable.

**Comment by tobiasdiez** [comment]
LGTM

**Pull request merged**